File: Recursion/Other_Examples/Factorial/Factorial.c
#include <stdio.h>

int factorial(int n) 
{
    if (n == 0) {
        return 1;
    } else {
        return n * factorial(n-1);
    }
}

int factorialIteration(int n) 
{
    int result = 1, i;

    for(i = 1; i <= n; i++) {
        result *= i;
    }

    return result;
}

int main(void)
{
    int r;

    printf("Result (recursion): %d\n", factorial(5));
    printf("Result (iteration): %d\n", factorialIteration(5));

    return 0;

}

File: Recursion/Java/factorial.java
public class factorial {

	public static void main(String[] args) {
		System.out.println(calcFactorial(6));
		System.out.println(calcFactorial(5));
		System.out.println(calcFactorial(4));
		System.out.println(calcFactorial(3));
		System.out.println(calcFactorial(2));
		System.out.println(calcFactorial(1));
	}

	public static int calcFactorial(int n) {
		if (n <= 1) {
			return 1;
		} else {
			return n * calcFactorial(n-1);
		}
	}

}
